src/damage.h

```diff
@@ -39,6 +40,17 @@ struct enum_traits<damage_type> {
     static constexpr damage_type last = damage_type::NUM;
 };
 
+struct barrel_desc {
+    units::length barrel_length;
+    float amount;
+
+    barrel_desc( units::length bl, float amt ) : barrel_length( bl ), amount( amt ) {}
+    void deserialize( const JsonObject &jo ) {
```

Comment:
Probably best to stick this in the .cpp file, so we don't have to put generic_factory.h in a header, whoops.